src/lib/util.ts

```diff
@@ -0,0 +1,55 @@
+import { SupportedNetwork, ReadQueryResult } from "../interfaces.js";
+
+declare let globalThis: any;
+
+// From https://www.npmjs.com/package/btoa
+const polyfills = {
+  btoa: function (str: string) {
+    const buffer = Buffer.from(str, "binary");
+
+    return buffer.toString("base64");
+  },
+};
+
+export const btoa = globalThis.btoa ?? polyfills.btoa;
+
+export const contractAddresses: { [index: string]: string } = {
+  // If building locally you can put your contract address here or use the contract connection option
+  localhost: "0xe7f1725e7734ce288f8367e1bb143e90bb3f0512",
```

Comment:
probably best to remove the placeholder string for localhost... could be empty string or some obvious placeholder, eg, "LOCAL_HARDHAT_CONTRACT". actually, could even come from env?

Reply:
We don't want to assume an ENV here at all, because this will also run in a browser. But the CLI can assume this and fill it in.

Reply:
It is only deterministic in that it is a fresh address and deploying the contract is the first thing you do with that address. But that might not always be the case.